ERCS/erc-7638.md

```diff
@@ -0,0 +1,190 @@
+
+## Motivation
+Typically, interactions between users and contracts involve a series of coherent operations, such as `approve-transferFrom`. While EOA wallets require users to confirm each operation sequentially, SCA wallets can confirm all operations with a single confirmation, completing all operations within a single call, thus achieving atomicity. If approve succeeds but transferFrom fails, it poses a security risk. The secure approach is to ensure that if one operation fails, all associated operations also fail, thereby ensuring atomicity. Therefore, we propose this encoding method to encode multiple parameters into bytes, compress on-chain data, and save gas. It can be used to implement both atomic and non-atomic operations.
+
+In addition to the atomic operation of `approve-transferFrom` mentioned above, gas payment delegation can also be achieved. It involves users and bundlers signing a set of calls, where the content of the calls includes:
+
+1. The user wishes to initiate multiple calls through his SCA.
+2. The user transfers 10 USDT to the bundler as fee, included within the calls.
+3. The bundler submits the calls, pay ETH gas and get the 10 USDT.
+
+The user encodes the content of the calls, attaches their signature to ensure its integrity, and sends it to the bundler. If the bundler considers the gas payment insufficient, they may choose not to submit it. However, if they approve the content of the calls, the signed transaction can be submitted. After execution, the user obtains the desired operations, and the bundler receives the fee.
+
+[EIP-4337](./eip-4337.md) also implements gas payment delegation. OCE and [EIP-4337](./eip-4337.md) are not mutually exclusive and can be implemented concurrently within an SCA.
+
+Based on empirical testing, OCE is simpler and more gas-efficient compared to alternative methods.
+
+## Specification
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+`Optimized Calls` comprises multiple `Optimized Call` bytes, each defined by the encoding of `to\value\data` as follows:
+
+### <center>|  to(20bytes)  |  value(32bytes)  |  data length(32bytes)  |  data(bytes)  |</center>
+
+Let:
+- `to`: The address of the called contract, corresponding to the Solidity address type, 20 bytes.
+- `value`: The amount of ETH(in wei) sent to the contract, in wei, corresponding to the Solidity uint type, 32 bytes.
+- `data length`: The length of the data(in bytes), corresponding to the Solidity uint type, 32 bytes.
+- `data`: The encoded functionData sent to the contract, corresponding to the Solidity bytes type, with a length defined by `data length`.
+
+Multiple `Optimized Call` units are concatenated to form an `Optimized Calls` sequence.
+
```

Comment:
I think your specification section needs a bit more information about who will be implementing this. Is this meant to be implemented by the dapp communicating directly with a SCA, or is this between the wallet browser extension and the SCA?

Might be overkill, but I do like diagrams (SVG if you can), if you you feel like making one!

Reply:
OK, added: This ERC **REQUIRED** SCA to be implemented in the contract, where the Dapp communicates with the SCA wallet extension to communicate the user's intentions to the wallet, which uses Batch Calls Encoding to send multiple calls as bytes to the user's SCA contract.
And made a markdown graph.